qringbuf.go

```diff
@@ -25,8 +221,21 @@ type Region struct {
 	qrb      *QuantizedRingBuffer
 }
 
-func (r *Region) Size() int     { return r.size }
+// Size returns the size of the region in bytes. It is equivalent to (but
+// cheaper than):
+//  len( r.Bytes )
+func (r *Region) Size() int { return r.size }
+
+// Bytes returns a slice of the underlying ring buffer. One should take care
+// to copy or finish using the returned slice before making another call to
+// NextRegion(…). If Reserve() has been invoked, then the slice is guaranteed
+// to remain intact before the corresponding Release() call.
 func (r *Region) Bytes() []byte { return r.qrb.buf[r.offset : r.offset+r.size] }
+
+// SubRegion is analogous to re-slicing. Supplying offset/length values causing
+// an out of bounds re-slice results in log.Panic(). As a special case one can
```

Comment:
If you want to be _really_ fast and sneaky, you could say that this is undefined behavior. If you did that, you'd be able to stick used regions into a `sync.Pool`.

Reply:
I don't see how... that can work? We need to chat about this more...

An aside: I did have an implementation with sync.Pool in dagger itself over millions of 9-byte slices, where the freeing is definitive and thus a Pool makes sense -  I did rip it all out, because while allocations went down in a profile, there was **no actual difference, not even sub-%, outright undetectable**

Spoke with Eric a bit, it seems the compiler just got far enough along that sync.Pool is becoming needless overhead.